Pod/Classes/KolodaView/KolodaView.swift

```diff
@@ -651,4 +513,114 @@ public class KolodaView: UIView, DraggableCardDelegate {
+        return insertedCards
+    }
+    
+    private func removeCards(cards: [DraggableCardView]) {
+        cards.forEach { card in
+            card.delegate = nil
+            card.removeFromSuperview()
+        }
+        visibleCards.removeLast(cards.count)
+    }
+    
+    private func removeCards(cards: [DraggableCardView], animated: Bool) {
+        if animated {
+            animator.applyRemovalAnimation(
+                cards,
+                completion: { _ in
+                    self.removeCards(cards)
+                }
+            )
+        } else {
+            self.removeCards(cards)
+        }
+    }
+    
+    public func insertCardAtIndexSet(indexRange: Range<Int>, animated: Bool = true) {
+        if let dataSource = dataSource {
+            let currentItemsCount = countOfCards
+            let visibleIndexes = [Int](indexRange).filter { $0 >= currentCardNumber && $0 < currentCardNumber + countOfVisibleCards }
+            let insertedCards = insertVisibleCardsWithIndexes(visibleIndexes.sort())
+            removeCards(visibleCards.dropFirst(countOfVisibleCards).map { $0 }, animated: animated)
+            for (index, card) in visibleCards.enumerate() {
+                card.alpha = shouldTransparentizeNextCard && index != 0 ? alphaValueSemiTransparent : alphaValueOpaque
+                card.userInteractionEnabled = index == 0
+            }
+            animator.resetBackgroundCards()
+            if animated {
+               animating = true
+                animator.applyInsertionAnimation(
+                    insertedCards,
+                    completion: { _ in
+                        self.animating = false
+                    }
+                )
+            }
+            
+            countOfCards = Int(dataSource.koloda(kolodaNumberOfCards: self))
+            assert(
+                currentItemsCount + indexRange.count == countOfCards,
+                "Cards count after update is not equal to data source count"
+            )
+        }
+    }
+    
+    // MARK: Cards managing - Deletion
+    
+    private func proceedDeletion(range: Range<Int>) {
+        let deletionIndexes = [Int](range)
+        deletionIndexes.sort { $0 > $1 }.forEach { deletionIndex in
+            let visibleCardIndex = deletionIndex - currentCardNumber
+            let card = visibleCards[visibleCardIndex]
+            card.delegate = nil
+            card.swipe(.Right)
+            visibleCards.removeAtIndex(visibleCardIndex)
+        }
+    }
+    
+    public func removeCardAtIndexRange(indexRange: Range<Int>, animated: Bool) {
+        if let dataSource = dataSource {
+            animating = true
+            let currentItemsCount = countOfCards
+            let visibleIndexes = [Int](indexRange).filter { $0 >= currentCardNumber && $0 < currentCardNumber + countOfVisibleCards }
+            if !visibleIndexes.isEmpty {
+                proceedDeletion(visibleIndexes[0]...visibleIndexes[visibleIndexes.count - 1])
```

Comment:
proceedDeletion(visibleIndexes[0]...visibleIndexes[visibleIndexes.count - 1])
could be
  proceedDeletion(visibleIndexes[0]..<visibleIndexes[visibleIndexes.count])